### PR TITLE
Refactor peering config, add test coverage

### PR DESCRIPTION
**Pull request by eculver**
This is a follow-up to #2 which added support for testing peering. This change refactors a bit of the config to remove some hard-coded string checks, to rename field names to clarify what they are tied to and then introduces a new field to expose a timeout that was hard-coded. It also adds some basic test coverage for the validation. 

Refs. NET-1127.

### CODE REVIEW

File: load/config.go [hunk 1]
```diff
@@ -12,16 +13,49 @@ const (
...
 	KV      *KVConfig
 	Peering *PeeringConfig
+	Target  Target
 }
 
 func (c *Config) Normalize() error {
+	if c.KV != nil && c.Peering != nil {
+		return errors.New("error validating configuration, cannot configure both KV and Peering, must choose one or the other")
+	}
+
```
Comment (huikang) [inline load/config.go:46]: are we going to allow both KV and peering to be not-nil? To make it consistent with the following two `if`, maybe we can return an error saying `target must be one of [kv, peering, ....]`

File: load/peering.go [hunk 2]
```diff
@@ -20,19 +20,36 @@ const (
...
-// peeringCluster represents a peering connection from the
-//
-//	acceptor to the peer cluster.
+// Normalize validates the Peering load test configuration. RegisterLimit and
+// NumServices must not be zero, otherwise, an error is returned.
+func (c *PeeringConfig) Normalize() error {
+	if c.RegisterLimit == 0.0 {
+		return fmt.Errorf("invalid RegisterLimit configuration: %v", c.RegisterLimit)
+	}
+	if c.NumServices == 0 {
```
Comment (huikang) [inline load/peering.go:43]: Actually `NumServices` could be 0 since the load test can be terminited by `-timeout`.

**Review by huikang — COMMENTED** [review]
@eculver , thanks for cleaning up the config. Left some minor comments about the usage.

**Comment by huikang** [comment]
@eculver , given that there is some unit test in the PR, will you be able to add the following workflow file to enable GHA with this PR?
```
name: Go

on:
  push:
    branches: [ "main" ]
  pull_request:
    branches: [ "main" ]

jobs:

  build:
    runs-on: ubuntu-latest
    steps:
    - uses: actions/checkout@v3

    - name: Set up Go
      uses: actions/setup-go@v3
      with:
        go-version: 1.18

    - name: Build
      run: go build -v ./...

    - name: Test
      run: go test -v ./...
```

File: load/config.go (hunk 1 above)
Comment (eculver) [reply load/config.go:46]: Yeah, the intent is for the `kv` and `peering` configs to be mutually exclusive. I think this is the case, right? I have a test for it below too named `"config.KV and config.Peering are mutually exclusive"`.

I think the message here is the same as what you are suggesting. Is this what you were suggesting or am I missing something?

File: load/peering.go (hunk 2 above)
Comment (eculver) [reply load/peering.go:43]: Got it. I'll update and add validation for it.

File: load/config.go (hunk 1 above)
Comment (huikang) [reply load/config.go:46]: Oh, I misunderstood the condition. It looks good.

**Review by huikang — APPROVED** [review]

**Pull request merged**